custom_components/tahoma/state_sensor.py

```diff
@@ -0,0 +1,105 @@
+"""Support for TaHoma sensors."""
+
+from dataclasses import dataclass
+from typing import Any, Callable, Optional, Union
+
+from homeassistant.components import sensor
+from homeassistant.const import PERCENTAGE, SIGNAL_STRENGTH_DECIBELS, VOLUME_LITERS
+from homeassistant.helpers.entity import Entity
+
+from .coordinator import TahomaDataUpdateCoordinator
+from .tahoma_entity import TahomaEntity
+
+
+@dataclass
+class StateDescription:
+    """Class to describe a sensor."""
+
+    key: str
+    name: str
+    icon: Optional[str] = None
+    unit: Union[None, str, Callable[[dict], str]] = None
+    value: Callable[[Any], Any] = lambda val: val
+    device_class: Optional[str] = None
+    default_enabled: bool = True
+
+
+supported_states = {
+    "core:BatteryState": StateDescription(
+        key="core:BatteryState",
```

Comment:
You can remove the key here and give it to the constructor. It will avoid to duplicate the data.

Reply:
Would you know how to pass this?